ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullFilterNode.java

```diff
@@ -159,13 +169,19 @@ public LogicalSchema getIntermediateSchema() {
     return intermediateSchema;
   }
 
-  private void validateWhereClause() {
+  private boolean validateWhereClause() {
+    boolean requiresTableScan = false;
     for (Expression disjunct : disjuncts) {
       final Validator validator = new Validator();
       validator.process(disjunct, null);
+      requiresTableScan = requiresTableScan || validator.requiresTableScan;
       if (!validator.isKeyedQuery) {
-        throw invalidWhereClauseException("WHERE clause missing key column for disjunct: "
-                + disjunct.toString(), isWindowed);
+        if (pullPlannerOptions.getTableScansEnabled()) {
+          requiresTableScan = true;
+        } else {
+          throw invalidWhereClauseException("WHERE clause missing key column for disjunct: "
```

Comment:
should we include in this error message that you can enable table scans using the config? (same for the below)